iac-extensions/arm/src/test/resources/checks/LogRetentionCheck/Microsoft.Network_networkWatchers_flowLogs/test.json

```diff
@@ -0,0 +1,99 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "resources": [
+    {
+      "name": "Raise Issue as days is too low",
+      "type": "Microsoft.Network/networkWatchers/flowLogs",
+      "apiVersion": "2022-07-01",
+      "properties": {
+        "retentionPolicy": {
+          "days": 7,
+          "enabled": true
+        }
+      }
+    },
+    {
+      "name": "Raise Issue as days is missing",
+      "type": "Microsoft.Network/networkWatchers/flowLogs",
+      "apiVersion": "2022-07-01",
+      "properties": {
+        "retentionPolicy": {
+          "enabled": true
+        }
+      }
+    },
+    {
+      "name": "Raise Issue as enabled is missing",
+      "type": "Microsoft.Network/networkWatchers/flowLogs",
+      "apiVersion": "2022-07-01",
+      "properties": {
+        "retentionPolicy": {
+          "days": 15
+        }
+      }
+    },
+    {
+      "name": "Raise Issue as enabled is false and days is too low",
+      "type": "Microsoft.Network/networkWatchers/flowLogs",
+      "apiVersion": "2022-07-01",
+      "properties": {
+        "retentionPolicy": {
+          "days": 7,
+          "enabled": false
+        }
+      }
+    },
+    {
+      "name": "Raise Issue as enabled is false",
+      "type": "Microsoft.Network/networkWatchers/flowLogs",
+      "apiVersion": "2022-07-01",
+      "properties": {
+        "retentionPolicy": {
+          "days": 15,
+          "enabled": false
+        }
+      }
+    },
+    {
+      "name": "Raise Issue as retentionPolicy is missing",
+      "type": "Microsoft.Network/networkWatchers/flowLogs",
+      "apiVersion": "2022-07-01",
+      "properties": {
+      }
+    },
+    {
+      "name": "Compliant",
+      "type": "Microsoft.Network/networkWatchers/flowLogs",
+      "apiVersion": "2022-07-01",
+      "properties": {
+        "retentionPolicy": {
+          "days": 15,
+          "enabled": true
+        }
+      }
+    },
+    {
+      "name": "Compliant",
+      "type": "Microsoft.Network/networkWatchers/flowLogs",
+      "apiVersion": "2022-07-01",
+      "properties": {
+        "retentionPolicy": {
+          "days": 0,
+          "enabled": true
+        }
+      }
+    },
+    {
+      "name": "Compliant",
+      "type": "Microsoft.Network/networkWatchers/flowLogs",
+      "apiVersion": "2022-07-01",
+      "properties": {
+        "retentionPolicy": {
+          "days": "wrong format for type",
+          "enabled": "wrong format for type"
+        }
+      }
+    }
```

Comment:
You can also add a test for different `type` and `flowLogs` as sub-resource (I know it's not in the ticket, but should work out of the box)